detection_rules/rule.py

```diff
@@ -604,10 +686,55 @@ def add_required_fields(self, obj: dict) -> None:
         if self.check_restricted_field_version(field_name=field_name):
             obj.setdefault(field_name, required_fields)
 
-    def add_setup(self, obj: dict) -> None:
+    def _add_setup(self, obj: dict) -> None:
         """Add restricted field setup to the obj."""
-        # field_name = "setup"
-        ...
+        rule_note = obj.get("note", "")
+        field_name = "setup"
+        field_value = obj.get(field_name)
+
+        if not self.check_explicit_restricted_field_version(field_name):
+            return
+
+        data_validator = self.data.data_validator
+
+        if data_validator.setup_in_note and data_validator.is_elastic_rule and not field_value:
+            parsed_note = self.data.parsed_note
+
+            # parse note tree
+            for i, child in enumerate(parsed_note.children):
+                if child.get_type() == "Heading" and "Setup" in gfm.render(child):
+                    field_value = self._get_setup_content(parsed_note.children[i + 1:])
+
+                    # clean up old note field
+                    investigation_guide = rule_note.replace("## Setup\n\n", "")
+                    investigation_guide = investigation_guide.replace(field_value, "").strip()
```

Comment:
There are some instances where the value is not removed from `note`, which means it is likely being rendered slightly different from the original within `_get_setup_content`

Reply:
It appears that the changes you made to use the marko github flavored markdown extension breaks the expected parsing. I'll see what we can do with still using `gfm`.
<img width="921" alt="Screen Shot 2022-07-14 at 4 56 09 PM" src="https://user-images.githubusercontent.com/1636709/179084449-fbbd16c7-6310-415c-8353-2185d9272649.png">
.

Reply:
Should be resolved with https://github.com/elastic/detection-rules/pull/2061/commits/836a25c881fcd1bbbe2fc1e3f3f0c3c6336d796f so im going to mark as resolved.